src/sagas/nanoContract.js

```diff
@@ -172,21 +171,37 @@ export function* registerNanoContract({ payload }) {
  * @param {string} after Transaction hash to start to get items
  * @param {Object} wallet Wallet instance from redux state
  *
+ * @description
+ * We use `after` because we look at time from present to future, therefore `after` means
```

Comment:
I don't get it... We are actually fetching transactions "before" a given transaction as the user "infinite scrolls" deeper, shouldn't we name the variable `before`?

Reply:
Sorry for the confusion, let me try again:
When we call `fetchHistory` with `after` param this denotes that we want new transactions _after_ a certain tx, and we are looking from the present moment toward the future. This is a common knowledge of how to think in time, _after_ means the future.

{txs that we have (past)} **{after (the last tx that we have)}** {txs we are interested in (future)}

However the method `getNanoContractHistory` uses the following signature:
`getNanoContractHistory(ncId, count, after, before)`.

Here, `after` denotes all the transactions from a certain tx towards the **past**. I believe it has something to do with the way we walk the dag. However, this way to frame time is not like the way we usually do, and it can makes things really odd in my opinion, and that's why I'm swapping the words `before` to `after`. After all, we are interested in the future when we ask a new chunk of transactions.

Reply:
The words `after` and `before` doesn't have any relation with future and past in this API. I think it's confusing to swap the words, we should use the same words as it's used in the API. `after` is the pagination parameter to get the history after a specific tx, and `before` the same. The API returns the transactions in the descending order, i.e., from the newest to the oldest, as it's expected in the screen.

Reply:
Ok. I just refactored it. Can you both take a new look at it?